src/helm/proxy/clients/auto_client.py

```diff
@@ -57,14 +58,24 @@ def _get_client(self, model: str) -> Client:
         client: Optional[Client] = self.clients.get(model)
 
         if client is None:
-            organization: str = model.split("/")[0]
-            cache_config: CacheConfig = self._build_cache_config(organization)
+
+            model_config: Optional[ModelConfig] = get_model_config(model)
+            if model_config:
+                model_type = model_config.model_type
+            else:
+                # TODO(#1502): Only use model_config.client_type to decide the client type,
+                # and don't fall back to organization. This requires populating the ModelConfig
+                # for every built-in model.
+                organization = model.split("/")[0]
+                model_type = organization
+
+            cache_config: CacheConfig = self._build_cache_config(model_type)
 
             if get_huggingface_model_config(model):
                 from helm.proxy.clients.huggingface_client import HuggingFaceClient
 
                 client = HuggingFaceClient(cache_config=cache_config)
-            elif organization == "openai":
+            elif model_type == "openai":
```

Comment:
I think the term 'model_type' is not exactly right - I think more of Transformer versus RNN.  Why not stick with organization?  It is really an organizational thing and has nothing to do with the underlying model.

Reply:
I think we can just make this a `ObjectSpec.class_name`. See the new example `model_deployments.yaml`.